File: app/src/main/java/com/gazilla/mihail/gazillaj/presentation/account/AccountView.java
package com.gazilla.mihail.gazillaj.presentation.account;

import com.arellomobile.mvp.InjectViewState;
import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.gazilla.mihail.gazillaj.utils.POJO.Success;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface AccountView  extends MvpView {

    void setUserInfo(String name, String phone, String email, String id);
    void showWorningDialog(String err);
    void showLoadingDialog();
    void clouseAppDialog();
}

File: app/src/main/java/com/gazilla/mihail/gazillaj/views/AccountView.kt
package com.gazilla.mihail.gazillaj.views

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(value = AddToEndSingleStrategy::class)
interface AccountView : MvpView {

    fun setUserId(id: String?)
    fun setUserInfo(name: String?, phone: String?, email:String?)

    fun showLoadDialog()

    fun closeLoadDialog()
    fun showMessageDialog(message : String?)


}